### PR TITLE
Erlang 21 support

### LINKED ISSUE #1396: Build with Erlang 21
## Expected Behavior

CouchDB should build and run with Erlang 21

## Current Behavior

Main repo and snappy dependency prevent building with Erlang 21

Also unit tests fails

## Possible Solution

Update deps and fix tests

### PR DESCRIPTION
**Pull request by nickva**
Erlang 21 support

Fixed #1396

### CODE REVIEW

**Comment by iilyak** [comment]
There is another place where monitored_by is used:
- https://github.com/apache/couchdb/blob/master/src/couch/test/couchdb_views_tests.erl#L526

**Comment by davisp** [comment]
+1 after some squerging

**Comment by nickva** [comment]
@iilyak good catch! 

@davisp good idea, will do some squerging

File: src/couch/src/couch_file.erl [hunk 1]
```diff
@@ -129,7 +129,7 @@ append_term_md5(Fd, Term, Options) ->
 
 append_binary(Fd, Bin) ->
     ioq:call(Fd, {append_bin, assemble_file_chunk(Bin)}, erlang:get(io_priority)).
-    
+
```
Comment (davisp) [inline src/couch/src/couch_file.erl]: Tsk tsk

**Pull request merged**